fipy/tools/timer.py

```diff
@@ -0,0 +1,63 @@
+import time
+
+class Timer(object):
+    """Context manager that measures time elapsed in context
+
+    Defaults to nanosecond precision (although probably only microsecond or
+    even millisecond accuracy).
+
```

Comment:
`examples.phase.simple` returns elapsed times in $[3\times10^5, 1\times10^8]$ ns.
In the logfile, these large numbers are difficult to read, esp. without thousands separators.
Alongside the accuracy vs. precision argument, I suggest scaling the timer's output from ns to μs or ms, making the unit string part of the Timer class instead of hard-coding, and possibly making the scaling factor and unit user-definable.

Reply:
They're not particularly intended to be human-readable. Anything that makes it easier for humans makes it harder for machine parsing. Stupid humans.

The units/type/magnitude of `Timer.elapsed` is a result of the output of the corresponding routine optionally passed to `timer=`. It defaults to [`time.perf_counter_ns()`](https://docs.python.org/3/library/time.html#time.perf_counter_ns). One of the issues with the previous way I was doing this (relying on the log time) was that it was only millisecond precise and had huge uncertainty, rendering useless any stats from small problems and lending quite a bit of uncertainty even to big ones. I was more interested in something unambiguously parsable than I was in something readable.